### ISSUE TITLE
Re-Order Category Folders

### ISSUE THREAD

**Issue by mikaelacaron** [mikaelacaron/brain-marks#67]
**Is your feature request related to a problem? Please describe.**
* No

**Describe the solution you'd like**
* Press and hold a row to reorder the categories

**Describe alternatives you've considered**
* None

**Additional context**
* None

**Comment by vcapilladeveloper** [comment]
@mikaelacaron I'm not making this issue, before start making it I normally review the task to understand and evaluate if I can make it.
Assign it to me this issue and I will try to find the best solution and explain to you before start to implement it.

**Comment by vcapilladeveloper** [comment]
@mikaelacaron hi, How you think about this: (I will try to fix one bug because I can't make longpress on Spacer() inside CategoryRow and I don't know why.)

https://user-images.githubusercontent.com/36862920/137602253-278ec21f-043c-449e-93c6-f05658a2dabe.MP4

:

**Comment by vcapilladeveloper** [comment]
Ok, I can end the Editing Mode for each movement, what do you think about it? Other issue that I'm trying to find the best solution is how the user edit the category? Maybe inside the category list view add an Edit button at the top right? 

To fix the problem on longpress inside an empty location from the row, I put this: 
```
CategoryRow(category: category)
    .contentShape(Rectangle())
    .onLongPressGesture {
        editMode = .active
    }
}
```
This applye the long press gesture to the entire CategoryRow instead of the content.

**Comment by mikaelacaron** [comment]
Hmmm that could work!
I originally based the design off the Notes app from Apple. To make it easy to use and understand.

I'm not sure if Notes is made with SwiftUI or UIKit.

Also you could change "Edit" to "Rename" instead, cause that's what Notes uses for verbiage

**Comment by vcapilladeveloper** [comment]
Great, I will make it and show you the result

**Comment by mikaelacaron** [comment]
That's okay! @vcapilladeveloper Do you want me to un-assign this issue from you? to allow others to contribute to this issue

**Comment by AC40** [comment]
Hey Mikaela, could you assign me to this issue? I would like to try and continue @vcapilladeveloper`s work.

**Comment by mikaelacaron** [comment]
Go for it! Thanks again for the help!

**Comment by AC40** [comment]
My pleasure. It's a great way to get to know more about the iOS world, other peoples code, conventions, etc.

**Comment by mikaelacaron** [comment]
Fixed with PR #111
